off-chain/src/Network/Types.purs

```diff
@@ -225,8 +228,9 @@ type LeaderStateInner a =
   , waitingForSignature :: Ref $ OrderedMap UUID Transaction
   , signatureResponses ::
       Queue.Queue (read :: Queue.READ, write :: Queue.WRITE)
-        (UUID /\ Either Unit Transaction)
+        (UUID /\ Maybe Transaction)
```

Comment:
Maybe make some sum type for this? I caught myself on trying to remember every time why we need type like this here.
Maybe something like `data SignResult = Signed Transaction | Rejected`

Reply:
it would need to include `TimedOut` (we don't distintion right now but we weren't focusing in that). Also, I don't think we should move more things in src/ and focus on test/